distributed/protocol/numpy.py

```diff
@@ -102,3 +102,37 @@ def deserialize_numpy_ndarray(header, frames):
                        strides=header['strides'])
 
         return x
+
+
+@dask_serialize.register(np.ma.core.MaskedArray)
+def serialize_numpy_maskedarray(x):
+    # Separate elements of the masked array that we need to deal with discretely.
+    data = x.data
+    mask = x.mask
+    fill_value = x.fill_value
+
+    # Make use of existing numpy serialization for the two ndarray elements of
+    # the masked array.
+    data_header, data_frames = serialize_numpy_ndarray(data)
+    mask_header, mask_frames = serialize_numpy_ndarray(mask)
+
+    header = {"data-header": data_header,
+              "mask-header": mask_header,
+              "fill_value": fill_value,
+              "nframes": [len(data_frames), len(mask_frames)]}
+    return header, data_frames + mask_frames
+
+
+@dask_deserialize.register(np.ma.core.MaskedArray)
+def deserialize_numpy_maskedarray(header, frames):
+    data_frames = frames[:header["nframes"][0]]
+    mask_frames = frames[header["nframes"][1]:]
```

Comment:
I think you actually only need `header['nframes'][0]` here in both cases.  I suspect that you only need this split point between the two lists of frames.  I don't think that you need both lengths.